src/entities/merkleDistributor.ts

```diff
@@ -0,0 +1,86 @@
+import keccak256 from 'keccak256';
+import { Contract, utils } from 'ethers';
+import { MerkleTree } from 'merkletreejs';
+import { EthConsts } from '@pendle/constants';
+import { contracts } from '../contracts';
+import { distributeConstantsByNetwork, isSameAddress, submitTransaction } from '../helpers';
+import { Token } from './token';
+import { TokenAmount } from './tokenAmount';
+import type { ChainSpecifics } from './types';
+
+// TODO: Annotate the contract with the appropriate type
+// import type { PendleMerkleDistributor } from '@pendle/core/typechain-types';
+
+export class PendleMerkleDistributor {
+  public readonly token: Token;
+  public readonly distributorAddress: string;
+
+  constructor({ address, chainId = EthConsts.common.CHAIN_ID }: { address: string; chainId: number }) {
+    const { contractAddresses } = distributeConstantsByNetwork(chainId);
+    const { merkleDistributors } = contractAddresses;
+    this.distributorAddress = merkleDistributors!.find(({ token }) => isSameAddress(token, address))!.distributor;
+    this.token = Token.find(address, chainId);
+  }
+
+  public static find({ address, chainId = EthConsts.common.CHAIN_ID }: { address: string; chainId: number }) {
+    return new PendleMerkleDistributor({ address, chainId });
+  }
+
+  // TODO: Complete this function to fetch merkle tree from backend server
+  public async merkleTree(): Promise<MerkleTree> {
```

Comment:
Please implement this function, assuming the file structure will be the same as https://github.com/pendle-finance/pendle-core-internal/blob/gas-saver/test/core/1_merkle-distributor.spec.ts#L35

Reply:
This has now been implemented according to the API in [https://github.com/pendle-finance/pendle-backend/pull/5](https://github.com/pendle-finance/pendle-backend/pull/5)